sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs

```diff
@@ -297,6 +298,15 @@ public virtual void ReceiveMessageStart(string identifier, int messageCount)
             }
         }
 
+        [Event(ReceiveMessageCanceledEvent, Level = EventLevel.Verbose, Message = "A receive operation was cancelled. (Identifier '{0}'). Error Message: '{1}'")]
+        public void ReceiveMessageCanceled(string identifier, string exception)
```

Comment:
Should we consider doing something similar for sending messages too?

Reply:
I considered this, but being that receiving is the only polling scenario we have, and the only scenario where customers have run into this issue, I didn't want to open up the can of worms of applying this pattern to every other operation. The main reason we need it for receiving is that functions, and potentially other third party libraries, will cancel the long polling operations when shutting down, and this may be transparent to the actual end user. I don't think it is likely that the same issue would come up for other types of operations.